### DISCUSSION TITLE
Anyone has idea about how to change certain CTK widget fonts?

### DISCUSSION THREAD

**Discussion by Gasiro** [TomSchimansky/CustomTkinter#2466] [Q&A] [answered]
I successfully changed the button fonts / label fonts, but have no idea how to change the font of 1.[CTkTabview tab font] and [CTkEntry placeholder_text font].
Anyone has any hint?
![Untitled-1](https://github.com/TomSchimansky/CustomTkinter/assets/117976338/b6fbacd4-229b-4b06-b7d5-9962a27f4231)

**Comment by dipeshSam** [comment, marked as answer]
@Gasiro 

There are two way to do so:
1. Set global font (Recommended)
2. Change font of Tabview's segmented button

We can change font of all the widgets just using one statement of `ThemeManager`. Otherwise, we can iterate through buttons, and can change their fonts individually.

Sample code for way first:
``` Python
from customtkinter import (
    CTk, 
    CTkTabview,
    CTkEntry,
    CTkFont
)

from customtkinter import ThemeManager

if __name__ == "__main__":
    app = CTk()
    
    # This will simply change font of ALL THE WIDGETS.
    ThemeManager.theme["CTkFont"] = CTkFont("Impact", 20)
    
    tabs = CTkTabview(app)
    tabs.add("Tab1")
    tabs.add("Tab2")
    tabs.add("Tab3")
    tabs.pack()
    
    entry = CTkEntry(app, placeholder_text="Write your comment...")
    entry.pack()

    app.mainloop()
```

Sample code if you only want specific widget to change:
``` Python 
from customtkinter import (
    CTk, 
    CTkTabview,
    CTkEntry,
    CTkFont
)

def apply_font_to_tabs(tabs: CTkTabview, font: CTkFont | tuple):
    """Applies given `font` either tuple or `CTkFont` to tab names."""
    tabs._segmented_button.configure(font=font)

if __name__ == "__main__":
    app = CTk()

    tabs = CTkTabview(app)
    tabs.add("Tab1")
    tabs.add("Tab2")
    tabs.add("Tab3")
    tabs.pack()

    # Use the function to iterate through.
    apply_font_to_tabs(tabs, CTkFont("impact", 15))
    
    # By default, font applies to both placeholder text and entry text.
    entry = CTkEntry(app, placeholder_text="Write your comments...", font=CTkFont("Impact", 15))
    entry.pack()

    app.mainloop()
```

**Both will produce output like:**
![ctk_fonts](https://github.com/TomSchimansky/CustomTkinter/assets/72375763/fcee9454-f875-4505-93a1-260383d48e80)


Hope, it will be helpful.
Regards

**Reply by Gasiro** [reply]
Thank you so much Dipesh. It looks much better now!
![WeChat2271a04a295a5a250e290464105ed0db](https://github.com/TomSchimansky/CustomTkinter/assets/117976338/46600cc3-770e-46d4-b5f5-849f88fb65b2)

**Reply by dipeshSam** [reply]
Yeah, that's amazing!
Keep it up.
